### DISCUSSION TITLE
Add backend to export database to a spreadsheet

### DISCUSSION THREAD

**Discussion by james-ball-qualcomm** [riscv-software-src/riscv-unified-db#234] [General]
Need to understand what data to export and in what format.

**Comment by AFOliveira** [comment]
@james-ball-qualcomm  Would this be to a table of instructions and CSRs?

**Comment by james-ball-qualcomm** [comment]
Actually, for this issue, it would just be a list of all parameters in a nice table. This is to support the current activity at RVI of creating a golden list of all parameters for all RVI standards. Currently the list is in this spreadsheet: https://docs.google.com/spreadsheets/d/1tFGLbocTp1YNn11aN8JTmvxL_23e7x8SlBNJfOPRmAE and it is under active development. We are proposing to do a one-time import of the spreadsheet into UDB and then be able to export it in a number of formats but the spreadsheet being the first. Exporting as a CSV might be acceptable but if there is some library available to export in xlsx format that might make a better result (not sure, just guessing).

Note that #227 was created by me since the spreadsheet has a category and we don't have that in UDB. We do have the extension(s) that define the parameter and that would be useful to export too. Also, I created #59 which I plan to work on soon. If you are thinking of creating a backend to export the parameters to a spreadsheet, you might want to wait for that issue to be completed (or help with it if you want).

**Comment by james-ball-qualcomm** [comment]
@dhower-qc adding

**Comment by AFOliveira** [comment]
If all you need is a script to parse the files and then create a set of different outputs from it, I can surely help! I think I'll wait for #59 to be complete, but I'll also take a look to see if I can help. 

> Exporting as a CSV might be acceptable but if there is some library available to export in xlsx format that might make a better result (not sure, just guessing)

Python allows xlsx, not sure about Ruby

**Comment by dhower-qc** [comment]
There is also an xlsx library for Ruby.

**Reply by james-ball-qualcomm** [reply]
Awesome! Maybe we could use that. Exporting to raw csv is painful to import directly into Excel.
